File: student-vault-backend/src/main/java/com/izzatkarimov/student_vault/service/StudentService.java
package com.izzatkarimov.student_vault.service;

import com.izzatkarimov.student_vault.dto.StudentDto;

import java.util.List;

public interface StudentService {

    // Create Student Service
    StudentDto createStudent(StudentDto studentDto);

    // Get Student Service
    StudentDto getStudentById(Long studentId);

    // Get All Students Service
    List<StudentDto> getAllStudents();

    // Update Student Service
    StudentDto updateStudent(Long studentId, StudentDto studentDto);

    // Delete Student
    void deleteStudent(Long studentId);

}

File: student-vault-frontend/src/services/StudentService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://18.197.121.196:8080/api/students';

export const listStudents = () => axios.get(REST_API_BASE_URL);

export const createStudent = (student) => axios.post(REST_API_BASE_URL, student);

export const getStudent = (studentId) => axios.get(REST_API_BASE_URL + '/' + studentId);

export const updateStudent = (studentId, student) => axios.put(REST_API_BASE_URL + '/' + studentId, student);

export const deleteStudent = (studentId) => axios.delete(REST_API_BASE_URL + '/' + studentId);